### PR TITLE
Upgrade agent-tweaker overlay image

### PR DESCRIPTION
**Pull request by knabben**
### What this PR does / why we need it
Upstream image cannot work so we need to overlay the agent tweaker image.

### Which issue(s) this PR fixes
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1331

### Describe testing done for PR
Running the version bump did not download the expected image.

### Release note

```release-note
NONE
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by blc1996 — APPROVED** [review]
LGTM, thanks for the fix

File: pkg/v1/providers/ytt/02_addons/cni/antrea/antrea_overlay.lib.yaml [hunk 1]
```diff
@@ -41,5 +41,7 @@ spec:
       initContainers:
         #@overlay/match by=overlay.subset({"name":"install-cni"})
         - image: #@ "{}/{}:{}".format(antreaImageRepo, antreaImage.imagePath, antreaImage.tag)
+        #@overlay/match by=overlay.subset({"name":"antrea-agent-tweaker"})
```
Comment (vijaykatam) [inline pkg/v1/providers/ytt/02_addons/cni/antrea/antrea_overlay.lib.yaml:44]: Shouldn't the image override come from kbld file in the package bundle? @blc1996 - Did we not deprecate imageRepo?

Comment (blc1996) [reply pkg/v1/providers/ytt/02_addons/cni/antrea/antrea_overlay.lib.yaml:44]: It is for initial deploy with CRS.

**Comment by knabben** [comment]
merged with #1345

**Pull request closed without merging**